js/list.controller.js

```diff
@@ -0,0 +1,91 @@
+angular.module('imdbApp')
+	.controller('listController', ['$scope', 'imdbService', '$location', '$mdDialog', function($scope, imdbService, $location, $mdDialog){
+
+		$scope.movies = [];
+		$scope.showGrid = false;
+		$scope.sort = 'rank';
+		imdbService.getImdbJson().success(function(data){
+			//generateCovers(data);
+			$scope.movies = data;
+		});
+
+		$scope.toggleGrid = function(){
+			if($scope.search.title) {
```

Comment:
we didn't discuss this in class but check out "ng-if" and "ng-show"

if you add that to an element, the element will only be shown if the condition is met.

For instance if you add ng-show to your grid element, you can just update the state of showGrid on a button click and the hide and show is taken care of for you. Little less code on your part. Try if it works?